staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -2212,3 +2216,189 @@ func TestApplySetParentValidation(t *testing.T) {
+					want := string(wantBytes)
+					if diff := cmp.Diff(want, got); diff != "" {
+						t.Errorf("GetObjects returned unexpected diff (-want +got):\n%s", diff)
+					}
+
+				}
+			})
+		})
+	}
+}
+
+func TestApplyObjects(t *testing.T) {
+	testdirs := []string{"testdata/prune/simple"}
+	for _, testdir := range testdirs {
+		t.Run(testdir, func(t *testing.T) {
+			k := fakekube.NewFakeKube(t)
+			f := util.NewFactory(k)
+
+			cmdtesting.WithAlphaEnvs([]cmdutil.FeatureGate{cmdutil.ApplySet}, t, func(t *testing.T) {
+				manifests := []string{"manifest1", "manifest2"}
+				for _, manifest := range manifests {
+					t.Logf("applying manifest %v", manifest)
+
+					cmd := &cobra.Command{}
+					flags := NewApplyFlags(genericclioptions.NewTestIOStreamsDiscard())
+					flags.AddFlags(cmd)
+					cmd.Flags().Set("filename", filepath.Join(testdir, manifest+".yaml"))
+					cmd.Flags().Set("applyset", filepath.Base(testdir))
+					cmd.Flags().Set("prune", "true")
+					cmd.Flags().Set("validate", "false")
+
+					o, err := flags.ToOptions(f, cmd, "kubectl", []string{})
+					if err != nil {
+						t.Fatalf("unexpected error creating apply options: %v", err)
+					}
+
+					// // TODO(justinsb): Enable validation once we unblock --applyset
+					// err = o.Validate()
+					// if err != nil {
+					// 	t.Fatalf("unexpected error from validate: %v", err)
+					// }
+
+					var unifiedOutput bytes.Buffer
+
+					o.Out = &unifiedOutput
+					o.ErrOut = &unifiedOutput
+
+					if err := o.Run(); err != nil {
+						t.Errorf("error running apply: %v", err)
+					}
+
+					got := unifiedOutput.String()
+
+					p := filepath.Join(testdir, manifest+"-expected-apply.txt")
+					wantBytes, err := os.ReadFile(p)
+					if err != nil {
+						t.Fatalf("error reading file %q: %v", p, err)
+					}
+					want := string(wantBytes)
+					if diff := cmp.Diff(want, got); diff != "" {
+						t.Errorf("GetObjects returned unexpected diff (-want +got):\n%s", diff)
+					}
+				}
+			})
+		})
+	}
+}
+
+func TestApplyFail(t *testing.T) {
+	testdirs := []string{"testdata/prune/simple"}
+	for _, testdir := range testdirs {
+		t.Run(testdir, func(t *testing.T) {
+			k := fakekube.NewFakeKube(t)
+			f := util.NewFactory(k)
+
+			cmdtesting.WithAlphaEnvs([]cmdutil.FeatureGate{cmdutil.ApplySet}, t, func(t *testing.T) {
+				manifests := []string{"manifest1", "manifest2"}
+				for i, manifest := range manifests {
+					if i != 0 {
+						k.AddBeforeDeleteHook(func(req *fakekube.DeleteRequest) *http.Response {
+							return &http.Response{StatusCode: http.StatusInternalServerError}
+						})
+					}
+					t.Logf("applying manifest %v", manifest)
+
+					var unifiedOutput bytes.Buffer
+					ioStreams := genericclioptions.IOStreams{
+						ErrOut: &unifiedOutput,
+						Out:    &unifiedOutput,
+						In:     bytes.NewBufferString(""),
+					}
+					cmdutil.BehaviorOnFatal(fatalNoExit(ioStreams))
+					defer cmdutil.DefaultBehaviorOnFatal()
+
+					cmd := NewCmdApply("kubectl", f, ioStreams)
```

Comment:
I wanted to highlight this way of invoking the commands, it feels more like a command line invocation (I suspect we could actually make a wrapper that looks even more like a command line invocation if we wanted to).

I did this because I wanted to be (more) certain that I was testing the actual output that would be printed (despite any cobra wrapping)...